### PR TITLE
Added publication date and author

### PR DESCRIPTION
**Pull request by nicosomb**
| Q             | A
| ------------- | ---
| Bug fix?      |  no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | yes
| Fixed tickets | https://github.com/wallabag/wallabag/issues/629
| License       | MIT

capture d ecran 2017-04-05 a 22 25 39

### CODE REVIEW

File: src/Wallabag/CoreBundle/Helper/ContentProxy.php [hunk 1]
```diff
@@ -79,6 +79,10 @@ public function updateEntry(Entry $entry, $url, array $content = [])
         $entry->setContent($html);
         $entry->setHttpStatus(isset($content['status']) ? $content['status'] : '');
 
+        if (isset($content['date']) && null !== $content['date']) {
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Helper/ContentProxy.php]: Maybe check if the value is empty too

File: src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig [hunk 2]
```diff
@@ -44,9 +44,15 @@
 
         <div id="article-informations">
             <i class="tool icon icon-calendar" title="{{ 'entry.view.created_at'|trans }}">
-                {{ entry.createdAt|date('Y-m-d') }}
+                {{ entry.createdAt|date('Y-m-d H:i') }}
             </i>
 
+            {%  if entry.publishedAt is not null %}
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig]: To much space before the if

File: src/Wallabag/CoreBundle/Resources/views/themes/material/Entry/entry.html.twig [hunk 3]
```diff
@@ -226,8 +226,14 @@
                 </li>
                 <li>
                     <i class="material-icons" title="{{ 'entry.view.created_at'|trans }}">today</i>
-                    {{ entry.createdAt|date('Y-m-d') }}
+                    {{ entry.createdAt|date('Y-m-d H:i') }}
+                </li>
+                {%  if entry.publishedAt is not null %}
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/material/Entry/entry.html.twig]: Same, too much space

File: src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig [hunk 4]
```diff
@@ -44,9 +44,15 @@
 
         <div id="article-informations">
             <i class="tool icon icon-calendar" title="{{ 'entry.view.created_at'|trans }}">
-                {{ entry.createdAt|date('Y-m-d') }}
+                {{ entry.createdAt|date('Y-m-d H:i') }}
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig]: We should use time_diff here.

File: tests/Wallabag/CoreBundle/Controller/EntryControllerTest.php [hunk 5]
```diff
@@ -138,6 +138,7 @@ public function testPostNewOk()
         $this->assertInstanceOf('Wallabag\CoreBundle\Entity\Entry', $content);
         $this->assertEquals($this->url, $content->getUrl());
         $this->assertContains('Google', $content->getTitle());
+        $this->assertEquals('2015-03-28 15:37:39', $content->getPublishedAt());
```
Comment (tcitworld) [inline tests/Wallabag/CoreBundle/Controller/EntryControllerTest.php]: DateTime != string issue.

File: src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig (hunk 4 above)
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig]: You mean something like "2 days ago"?

File: src/Wallabag/CoreBundle/Helper/ContentProxy.php [hunk 6]
```diff
@@ -79,10 +79,14 @@ public function updateEntry(Entry $entry, $url, array $content = [])
         $entry->setContent($html);
         $entry->setHttpStatus(isset($content['status']) ? $content['status'] : '');
 
-        if (isset($content['date']) && null !== $content['date']) {
+        if (isset($content['date']) && null !== $content['date'] && '' !== $content['date']) {
             $entry->setPublishedAt(new \DateTime($content['date']));
         }
 
+        if (isset($content['authors']) && null !== $content['authors'] && [] !== $content['authors']) {
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Helper/ContentProxy.php]: Could you try using only `!empty($content['authors'])` in the whole `if`?

File: src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig [hunk 7]
```diff
@@ -47,12 +47,20 @@
...
             </i>
 
-            {%  if entry.publishedAt is not null %}
+            {% if entry.publishedAt is not null %}
                 <i class="tool icon icon-pencil2" title="{{ 'entry.view.published_at'|trans }}">
                     {{ entry.publishedAt|date('Y-m-d H:i') }}
                 </i>
             {% endif %}
 
+            {% if entry.publishedBy is not null and entry.publishedBy is not empty %}
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig]: The second condition is enough I guess

File: src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig (hunk 4 above)
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig]: https://twig-extensions.readthedocs.io/en/latest/date.html#time-diff

Comment (tcitworld) [inline src/Wallabag/CoreBundle/Resources/views/themes/baggy/Entry/entry.html.twig]: But that would be another PR.

File: app/DoctrineMigrations/Version20170405182620.php [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        $entryTable = $schema->getTable($this->getTable('entry'));
+
+        $this->skipIf($entryTable->hasColumn('published_at'), 'It seems that you already played this migration.');
```
Comment (j0k3r) [inline app/DoctrineMigrations/Version20170405182620.php]: You should add a test on `published_by` too, nope?

Comment (nicosomb) [inline app/DoctrineMigrations/Version20170405182620.php]: Done.
